AVT-990 [Spectron][Windows] Add test "Verify Always on Top options when multiple applications are opened"

thaisym1912 opened the pull request:
## Description
Add test "Verify Always on Top options when multiple applications are opened" [AVT-990](https://perzoinc.atlassian.net/browse/AVT-9990)

## Run
Please refer attachment for more details
[Jest Test Report.zip](https://github.com/symphonyoss/SymphonyElectron/files/2204279/Jest.Test.Report.zip)

vantruongt2 commented on tests/spectron/alwaysOnTop.spectron.js:
```diff
@@ -1,190 +1,65 @@
+    it('Verify Always on Top options when multiple applications are opened', async (done) => {
+        try {
+            await webActions.minimizeWindows();
+            await windowActions.openAppInMaximize("C:\\Windows\\notepad.exe");
```
Should move `openAppInMaximize` method to other.

thaisym1912 replied:
Updated

vantruongt2 commented on tests/spectron/alwaysOnTop.spectron.js:
```diff
@@ -1,190 +1,65 @@
+        try {
+            await webActions.minimizeWindows();
+            await windowActions.openAppInMaximize("C:\\Windows\\notepad.exe");
+            await windowActions.openAppInMaximize("C:\\Windows\\system32\\mspaint.exe");
```
Same here

thaisym1912 replied:
Updated

vantruongt2 commented on tests/spectron/spectronSetup.js:
```diff
@@ -26,22 +27,36 @@ class App {
+    async startApplication(configurations) {
+        try {
+            this.app = await this.app.start();
+            await this.app.client.waitForVisible(ui.TITLE_BAR, 60000);
```
Should define the variable for TIME_OUT

thaisym1912 replied:
Updated

vantruongt2 commented on tests/spectron/spectronWindowsActions.js:
```diff
@@ -0,0 +1,80 @@
+        return this.app.browserWindow.getPosition();
+    }
+
+    async setPosition(x, y) {
```
Remove x,y argument if you don't use them

thaisym1912 replied:
Updated

VishwasShashidhar commented:
@vantruongt2 Please rebase this PR.

thaisym1912 commented:
@VishwasShashidhar Completed rebase

thaisym1912 commented:
Updated the test with Menu selection and added results
[Jest Test Report.zip](https://github.com/symphonyoss/SymphonyElectron/files/2218222/Jest.Test.Report.zip)

KiranNiranjan commented on package.json:
```diff
@@ -109,6 +112,7 @@
   },
   "dependencies": {
     "@paulcbetts/system-idle-time": "1.0.4",
+    "applescript": "^1.0.0",
```
This should be a devDependency

KiranNiranjan commented on tests/spectron/spectronUtils.js:
```diff
@@ -1,8 +1,26 @@
 const childProcess = require('child_process');
+const { isMac } = require('../../js/utils/misc.js');
+// const applescript = require('applescript');
```
Remove commented out code

KiranNiranjan commented on tests/spectron/spectronUtils.js:
```diff
@@ -1,8 +1,26 @@
+                    });
+                }
+            });
+            //await applescript.execString('tell application "TextEdit" tell window 1 set zoomed to true end tell end tell');
```
same here
